Bots/SunBot.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using WeatherMonitoringAndReportingService.BotConfigurations;
+using WeatherMonitoringAndReportingService.Interfaces;
+
+namespace WeatherMonitoringAndReportingService.Bots
+{
+    public class SunBot : IWeatherBot
+    {
+        private readonly BotConfiguration _configuration;
+
+        public SunBot(BotConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public string Message => _configuration.Message;
+        public bool Enabled => _configuration.Enabled;
+
+        public void CheckWeather(double humidity, double temperature)
+        {
+            try
+            {
+                if (_configuration != null && _configuration.Enabled && _configuration.TemperatureThreshold.HasValue && temperature >= _configuration.TemperatureThreshold)
+                {
+                    Console.WriteLine(_configuration.Message);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"An error occurred while checking weather for SunBot: {ex.Message}");
+            }
+        }
```

Comment:
after having base Bot class then this method should be shared between sun and rain and override in snow